src/core/reference/include/openvino/reference/paged_attention.hpp

```diff
@@ -0,0 +1,330 @@
+                    } else {
+                        int32_t kv_head = h % num_kv_heads;
+                        const T* key_vec =
+                            key_cache + (((block_id * num_kv_heads + kv_head) * block_size + token_offset) * head_size);
+
+                        // Check for RoPE adjustment.
+                        bool do_rotate = false;
+                        int32_t rotated_index = -1;
+                        if (rotated_block_indices && rotation_deltas && rotation_trig_lut) {
+                            do_rotate = paged_attention_utils::should_rotate(block_id,
+                                                                             rotated_block_indices,
+                                                                             rotated_block_indices_shape,
+                                                                             rotated_index);
+                        }
+                        if (do_rotate) {
+                            int32_t trig_index = paged_attention_utils::get_trig_index(rotation_deltas,
+                                                                                       rotation_deltas_shape,
+                                                                                       rotated_index,
+                                                                                       token_offset,
+                                                                                       block_size);
+                            std::vector<T> temp_key(key_vec, key_vec + head_size);
+                            paged_attention_utils::apply_rope(temp_key.data(),
+                                                              head_size,
+                                                              rotation_trig_lut,
+                                                              trig_index);
+                            score_val = paged_attention_utils::dot_product(q_vec, temp_key.data(), head_size);
+                        } else {
+                            score_val = paged_attention_utils::dot_product(q_vec, key_vec, head_size);
+                        }
+                    }
+                } else {
+                    // Retrieve key from new input.
+                    int32_t new_token_idx = subsequence_begins ? (subsequence_begins[seq_idx] + (k - seq_past_tokens))
+                                                               : (k - seq_past_tokens);
+                    int32_t kv_head = h % num_kv_heads;
+                    const T* key_vec = key + new_token_idx * kv_shape[1] + kv_head * head_size;
+                    score_val = paged_attention_utils::dot_product(q_vec, key_vec, head_size);
+                }
+                // Apply scale and add alibi bias.
+                score_val *= scale;
+                T alibi = alibi_slopes ? alibi_slopes[h % num_kv_heads] : T(0);
+                score_val += alibi * static_cast<T>(k);
```

Comment:
According to the [paper](https://arxiv.org/pdf/2108.12409), it seems it's not correct for alibi? The formula:
`softmax(q * transpose(k) + m · [−(i − 1), ..., −2, −1, 0])`

Reply:
You're right, changed it now!